### ISSUE TITLE
Problem syncing Stripe subscription: " no implicit conversion of Array into String (TypeError)"

### ISSUE THREAD

**Issue by marckohlbrugge** [pay-rails/pay#829]
Calling `.sync!` on one of my `Pay::Subscription`'s is raising this error:

```ruby
irb(main):010:0> Pay::Subscription.find_by(processor_id: "sub_foobar123").sync!
/[…]/gems/activerecord-7.0.5/lib/active_record/store.rb:230:in `[]=': no implicit conversion of Array into String (TypeError)
```

Here's the full stack trace:

```
gems/activerecord-7.0.4/lib/active_record/store.rb:230:in `[]=': no implicit conversion of Array into String (TypeError)
    from gems/activerecord-7.0.4/lib/active_record/store.rb:230:in `write'
    from gems/activerecord-7.0.4/lib/active_record/store.rb:245:in `write'
    from gems/activerecord-7.0.4/lib/active_record/store.rb:213:in `write_store_attribute'
    from gems/activerecord-7.0.4/lib/active_record/store.rb:136:in `block (3 levels) in store_accessor'
    from gems/activemodel-7.0.4/lib/active_model/attribute_assignment.rb:49:in `public_send'
    from gems/activemodel-7.0.4/lib/active_model/attribute_assignment.rb:49:in `_assign_attribute'
    from gems/activerecord-7.0.4/lib/active_record/attribute_assignment.rb:21:in `block in _assign_attributes'
    from gems/activerecord-7.0.4/lib/active_record/attribute_assignment.rb:13:in `each'
    from gems/activerecord-7.0.4/lib/active_record/attribute_assignment.rb:13:in `_assign_attributes'
    from gems/activemodel-7.0.4/lib/active_model/attribute_assignment.rb:34:in `assign_attributes'
    from gems/activerecord-7.0.4/lib/active_record/persistence.rb:779:in `block in update!'
    from gems/activerecord-7.0.4/lib/active_record/transactions.rb:354:in `block in with_transaction_returning_status'
    from gems/activerecord-7.0.4/lib/active_record/connection_adapters/abstract/database_statements.rb:314:in `transaction'
    from gems/activerecord-7.0.4/lib/active_record/transactions.rb:350:in `with_transaction_returning_status'
    from gems/activerecord-7.0.4/lib/active_record/persistence.rb:778:in `update!'
    from gems/pay-6.6.1/lib/pay/stripe/subscription.rb:96:in `block in sync'
```

And here are all the attributes 

```ruby
sub = ::Stripe::Subscription.retrieve({id: subscription_id}.merge(expand_options), {stripe_account: nil}.compact)
irb(main):044:0> sub.each { |k, v| puts [k, v.class].join(": ") }; nil
id: String
object: String
application: NilClass
application_fee_percent: NilClass
automatic_tax: Stripe::StripeObject
billing_cycle_anchor: Integer
billing_thresholds: NilClass
cancel_at: NilClass
cancel_at_period_end: FalseClass
canceled_at: NilClass
cancellation_details: Stripe::StripeObject
collection_method: String
created: Integer
currency: String
current_period_end: Integer
current_period_start: Integer
customer: String
days_until_due: NilClass
default_payment_method: NilClass
default_source: NilClass
default_tax_rates: Array
description: NilClass
discount: NilClass
ended_at: NilClass
items: Stripe::ListObject
latest_invoice: Stripe::Invoice
livemode: TrueClass
metadata: Stripe::StripeObject
next_pending_invoice_item_invoice: NilClass
on_behalf_of: NilClass
pause_collection: NilClass
payment_settings: Stripe::StripeObject
pending_invoice_item_interval: NilClass
pending_setup_intent: NilClass
pending_update: NilClass
plan: Stripe::Plan
quantity: Integer
schedule: NilClass
start_date: Integer
status: String
test_clock: NilClass
transfer_data: NilClass
trial_end: Integer
trial_settings: Stripe::StripeObject
trial_start: Integer
=> nil
```

I'm on Stripe API version `2022-11-15` and `pay` version `6.6.1`

**Comment by excid3** [comment]
Hmm, I am not entirely sure what's going on here. I might have to have you DM me the details to see if I can spot the issue.

Unless you can create an example that I can reproduce?

**Comment by marckohlbrugge** [comment]
I think I found it!

Narrowed it down to the `subscription_items` attribute. I think when trying to sync up my existing data, I set the `data` column for all `pay_subscriptions` to a full JSON object of the Stripe Subscription. Somehow, this prevented the sync method from working correctly. I've now nullified all `data` columns and called `sync` on each subscription. That seems to have solved it 👍
